sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegen.scala

```diff
@@ -284,7 +284,14 @@ case class WholeStageCodegen(plan: CodegenSupport, children: Seq[SparkPlan])
         ${ctx.declareAddedFunctions()}
 
         protected void processNext() throws java.io.IOException {
-          ${code.trim}
+          try {
+            ${code.trim}
+          } catch (final Throwable e) {
+            org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(this.getClass());
+            logger.error("The method processNext() is generated for " +
+                         "${plan.origin.callSite.getOrElse("unknown")}");
+            throw e;
+          }
```

Comment:
While I generated ```class GeneratedIterator extends org.apache.spark.sql.execution.BufferedRowIterator implements org.apache.spark.Logging {```, I got the error ```org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 11, Column 6: Non-abstract class "org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator" must implement method "public abstract org.slf4j.Logger org.apache.spark.Logging.org$apache$spark$Logging$$log_()"```.

According to this [entry](http://stackoverflow.com/questions/7637752/using-scala-traits-with-implemented-methods-in-java), we may support ```Logging``` for now. But, it seems too fragile regarding to future changes in ```Logging```. As a result, I would like to use ```Logger```.